### PR TITLE
HHH-12297 - Relations are not loaded when using Fetch Profiles

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-12297

### CODE REVIEW

**Comment by ladislavovic** [comment]
It is OK according to me. I also used it in my project integration tests (which also test fetch profiles) and it passed.

File: hibernate-core/src/test/java/org/hibernate/test/fetchprofiles/EntityLoadedInTwoPhaseLoadTest.java [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+import org.hibernate.annotations.FetchMode;
+import org.hibernate.annotations.FetchProfile;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.cfg.Environment;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Test;
+
+import static com.sun.tools.internal.ws.wsdl.parser.Util.fail;
```
Comment (breglerj) [inline hibernate-core/src/test/java/org/hibernate/test/fetchprofiles/EntityLoadedInTwoPhaseLoadTest.java]: Is this intentional? Shouldn't this be `import static org.junit.Assert.fail`?

Comment (vladmihalcea) [reply hibernate-core/src/test/java/org/hibernate/test/fetchprofiles/EntityLoadedInTwoPhaseLoadTest.java]: Thanks for noticing it. I'll fix it.

Comment (breglerj) [reply hibernate-core/src/test/java/org/hibernate/test/fetchprofiles/EntityLoadedInTwoPhaseLoadTest.java]: While you’re fixing it, could you please also quote the entity name of the entity “Start”? That’ll fix the test on HANA.

Comment (vladmihalcea) [reply hibernate-core/src/test/java/org/hibernate/test/fetchprofiles/EntityLoadedInTwoPhaseLoadTest.java]: Sure thing.

**Pull request merged**